syntax = "proto3";

package ServiceInfo;

enum Status
{
    UNSPECIFIED = 0;
    OK = 1;
    FALSE = 2;
}

message MicroServiceInfo
{
    string service_name = 1;
    string ip = 2;
    string port = 3;
    string version = 4;
    Status status = 5;
}

message GetServiceInfoReq
{
    string service_name = 1;
}

message GetServiceInfoResp
{
    MicroServiceInfo service_info = 1;
}

message SetServiceInfoReq
{
    MicroServiceInfo service_info = 1;
}

message SetServiceInfoResp
{
    Status status = 1;
}

service ServiceRegistry
{
    rpc getServiceInfo(GetServiceInfoReq) returns (GetServiceInfoResp){}
    rpc setServiceInfo(SetServiceInfoReq) returns (SetServiceInfoResp){}
}